File: LeetCode/442-find-all-duplicates-in-an-array/ex_442_dummy.java
public class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> res = new ArrayList<>();
        HashSet<Integer> hSet = new HashSet<>();
        for (int n : nums) {
            if (hSet.contains(n))
                res.add(n);
            else
                hSet.add(n);
        }
        return res;
    }
}

File: LeetCode/442-find-all-duplicates-in-an-array/ex_442_dummy.py
class Solution(object):
    def findDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        mset = set()
        res = []
        for n in nums:
            if n in mset:
                res.append(n)
            else:
                mset.add(n)
        return res